x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/action/cache/TransportSearchableSnapshotsNodeCachesStatsAction.java

```diff
@@ -0,0 +1,270 @@
+            super(nodes);
+        }
+
+        public NodesRequest(StreamInput in) throws IOException {
+            super(in);
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+        }
+    }
+
+    public static class NodeCachesStatsResponse extends BaseNodeResponse implements ToXContentFragment {
+
+        private final int numRegions;
+        private final long regionSize;
+        private final long writes;
+        private final long bytesWritten;
+        private final long reads;
+        private final long bytesRead;
+        private final long evictions;
+
+        public NodeCachesStatsResponse(
+            DiscoveryNode node,
+            int numRegions,
+            long regionSize,
+            long writes,
+            long bytesWritten,
+            long reads,
+            long bytesRead,
+            long evictions
+        ) {
+            super(node);
+            this.numRegions = numRegions;
+            this.regionSize = regionSize;
+            this.writes = writes;
+            this.bytesWritten = bytesWritten;
+            this.reads = reads;
+            this.bytesRead = bytesRead;
+            this.evictions = evictions;
+        }
+
+        public NodeCachesStatsResponse(StreamInput in) throws IOException {
+            super(in);
+            this.numRegions = in.readVInt();
+            this.regionSize = in.readVLong();
+            this.writes = in.readVLong();
+            this.bytesWritten = in.readVLong();
+            this.reads = in.readVLong();
+            this.bytesRead = in.readVLong();
+            this.evictions = in.readVLong();
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeVInt(numRegions);
+            out.writeVLong(regionSize);
+            out.writeVLong(writes);
+            out.writeVLong(bytesWritten);
+            out.writeVLong(reads);
+            out.writeVLong(bytesRead);
+            out.writeVLong(evictions);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject(getNode().getId());
+            {
+                builder.startObject("shared_cache");
+                {
+                    builder.field("reads", reads);
+                    builder.humanReadableField("bytes_read_in_bytes", "bytes_read", ByteSizeValue.ofBytes(bytesRead));
+                    builder.field("writes", writes);
+                    builder.humanReadableField("bytes_written_in_bytes", "bytes_written", ByteSizeValue.ofBytes(bytesWritten));
+                    builder.field("evictions", evictions);
+                    builder.field("num_regions", numRegions);
+                    builder.humanReadableField("size_in_bytes", "size", ByteSizeValue.ofBytes(numRegions * regionSize));
```

Comment:
let's show the size that the user specified, not the rounded down one

Reply:
Ok, I pushed 7d0c96f1abef13551a8c5051171baa2df844cfc4